packages/blockchain/src/index.ts

```diff
@@ -128,14 +136,30 @@ export default class Blockchain implements BlockchainInterface {
       })
     }
 
-    this._validatePow = opts.validatePow !== undefined ? opts.validatePow : true
+    this._validateConsensus = opts.validateConsensus !== undefined ? opts.validateConsensus : true
     this._validateBlocks = opts.validateBlocks !== undefined ? opts.validateBlocks : true
 
     this.db = opts.db ? opts.db : level()
     this.dbManager = new DBManager(this.db, this._common)
 
-    if (this._validatePow) {
-      this.ethash = new Ethash(this.db)
+    if (this._validateConsensus) {
+      if (this._common.consensusType() !== 'pow') {
+        throw new Error('consensus validation only supported for pow chains')
+      }
+      if (this._common.consensusAlgorithm() !== 'ethash') {
+        throw new Error('consensus validation only supported for pow ethash algorithm')
+      }
+
+      this._ethash = new Ethash(this.db)
+    }
+
+    if (this._validateBlocks) {
+      if (this._common.consensusType() !== 'pow') {
+        throw new Error('block validation only supported for pow chains')
+      }
+      if (this._common.consensusAlgorithm() !== 'ethash') {
```

Comment:
These checks are at multiple places, we might want to create a helper function for these tests?

Reply:
I wanted to have slightly different error messages depending on where the errors occur, that's the reasoning for not having this unified.